modules/calendar/search.php

```diff
@@ -0,0 +1,173 @@
+        </div>
+      </div>
+      <div class="form-group">
+        <label class="control-label col-sm-2" for="startDate">Start</label>
+        <div class="col-sm-4">
+          <input type="text" class="form-control" id="startDate" name="startDate" placeholder="Date" required value="<?php echo $_POST['startDate'];?>">
+        </div>
+        <label class="control-label col-sm-2" for="endDate">End</label>
+        <div class="col-sm-4">
+          <input type="text" class="form-control" id="endDate" name="endDate" placeholder="Date" required value="<?php echo $_POST['endDate'];?>">
+        </div>
+      </div>
+      <div class="form-group">
+        <label class="control-label col-sm-2" for="provider_id">Provider</label>
+        <div class="col-sm-10">
+          <select class="form-control" id="provider_id" name="provider_id">
+            <option value="0">All Providers</option>
+            <?php
+            foreach($providers as $provider) {
+              $id = $provider['id'];
+              $selected = ($_POST['provider_id'] == $id) ? "selected" : "";
+              echo "<option value='$id' $selected>" . htmlspecialchars($provider['lname'],ENT_QUOTES) . ", " . htmlspecialchars($provider['fname'],ENT_QUOTES) . "</option>\n";
+            }
+            ?>
+          </select>
+        </div>
+      </div>
+      <div class="form-group">
+        <label class="control-label col-sm-2" for="facility_id">Facility</label>
+        <div class="col-sm-10">
+          <select class="form-control" id="facility_id" name="facility_id">
+            <option value="0">All Facilities</option>
+            <?php
+            foreach($facilities as $facility) {
+              $id = $facility['id'];
+              $selected = ($_POST['facility_id'] == $id) ? "selected" : "";
+              echo "<option value='$id' $selected>" . htmlspecialchars($facility['name'],ENT_QUOTES) . "</option>\n";
+            }
+            ?>
+          </select>
+        </div>
+      </div>
+      <div class="form-group"> 
+        <div class="col-sm-offset-2 col-sm-10">
+          <button type="submit" class="btn btn-default" name="submit" value="1">Submit</button>
+        </div>
+      </div>
+    </form>
+    
+    <!-- if search performed -->
+    <?php if($_SERVER['REQUEST_METHOD'] == 'POST') { 
+      $pid = ($_POST['provider_id'] == "0") ? null : $_POST['provider_id'];
+      $fid = ($_POST['facility_id'] == "0") ? null : $_POST['facility_id'];
+      $appointments = fetchAppointments(fixDate($_POST['startDate']), fixDate($_POST['endDate']), null, $pid, $fid);
+      
+      // filter the appointments using the keyword
+      $filteredAppointments = array_filter($appointments, "keywordFilter");
+    ?>
+    <table class="table table-hover">
+      <tr>
+        <th>Date</th>
+        <th>Time</th>
+        <th>Provider</th>
+        <th>Patient</th>
+        <th>Category</th>
```

Comment:
use this on the above information to ensure translation.
`<th><?php echo xlt('Category')?></th>`
Do it for date,time,provider and patient also